File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<long long, int> pref;
        pref[0] = 1;
        long long sum = 0;

        int ans = 0;
        for(int i = 0; i<nums.size(); ++i){
            sum += nums[i];
            if(pref.find(sum - k) != pref.end())   ans += pref[sum - k];
            pref[sum]++;
        }
        return ans;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ans = 0
        pref = {}
        pref[0] = 1
        sum = 0
        for i in nums:
            sum += i
            if (sum-k) in pref:
                ans += pref[sum - k]
            pref[sum] = pref.get(sum, 0) + 1
        return ans